BetterShardsBungee/src/main/java/vg/civcraft/mc/bettershards/bungee/BungeeDatabaseHandler.java

```diff
@@ -133,7 +111,7 @@ public String getServerName(UUID uuid) {
 			ResultSet set = getServer.executeQuery();
 			if (!set.next())
 				return null;
-			return set.getString(1);
+			return ProxyServer.getInstance().getServerInfo(set.getString(1));
```

Comment:
Any reason you don't drop that into the cache before returning it? That way this method is "cache-optimized" on re-entry -- cache miss leads to lookup leads to cache save.